index.js

```diff
@@ -0,0 +1,140 @@
+let selectedRegion = null;
+let countries= null;
+const container = document.querySelector('.countries-grid');
+
+document.addEventListener('DOMContentLoaded', () => {
+    const dropdownWrapper = document.querySelector('.dropdown-wrapper');
+    const dropdownHeader = dropdownWrapper.querySelector('.dropdown-header');
+    const dropdownBody = dropdownWrapper.querySelector('.dropdown-body');
+    //open the dropdown
+    dropdownHeader.addEventListener('click', (event) => {
+        dropdownWrapper.classList.toggle('open');
+    });
+    //check if need to close the dropdown
+    document.addEventListener('click', (event) => {
+        if (dropdownWrapper.classList.contains('open') && !dropdownWrapper.contains(event.target)) {
+            dropdownWrapper.classList.remove('open');
+        }
+    });
+    //get the region that choosen
+    dropdownBody.addEventListener('click', (event) => {
+        const clickedElement = event.target;
+
+        if(clickedElement.tagName === 'LI')
+        {
+            selectedRegion= clickedElement.dataset.region;
+            console.log(selectedRegion);
+        }
+        // Re-render countries based on selected region
+        renderCountries(countries, container);
+    });
+  });
+
+
+function createCountryCard(country) {
+    // Create the main <a> element
+    const card = document.createElement('a');
+    card.className = 'country scale-effect';
+    card.href = '#';
+    card.dataset.countryName = country.name;
+  
+    // Create the flag container and image
+    const flagDiv = document.createElement('div');
+    flagDiv.className = 'country-flag';
+  
+    const flagImg = document.createElement('img');
+    flagImg.src = country.flag;
+    flagImg.alt = `${country.name} Flag`;
+  
+    flagDiv.appendChild(flagImg);
+  
+    // Create the info container
+    const infoDiv = document.createElement('div');
+    infoDiv.className = 'country-info';
+  
+    const countryTitle = document.createElement('h2');
+    countryTitle.className = 'country-title';
+    countryTitle.textContent = country.name;
+  
+    infoDiv.appendChild(countryTitle);
+  
+    // Create the ul
+    const ulList = document.createElement('ul');
+    ulList.className = 'country-brief';
+  
+    const populationItem = document.createElement('li');
+    populationItem.innerHTML = `<strong>Population: </strong>${country.population}`;
```

Comment:
Avoid using `innerHTML`. 

https://medium.com/@jasen.miyamoto/appending-to-the-dom-and-the-dangers-of-using-innerhtml-debd77e53e70